### PR TITLE
Saichallenger infra for func tests

### PR DESCRIPTION
**Pull request by chrispsommers**
Adding directories and basic test resources (setup files, etc.) for DASH functional tests using SAI Challenger framework.
- Add directory `test/test-cases/functional/saic` based on existing `test/test-cases/scale/saic`
- New make targets `run-saichallenger-functional-tests`, `run-saichallenger-scale-tests`
- New directory `test/test-cases/utils` in anticipation of new test cases WIP.
- Docker volume mounts include new directories
- Interim change to `test/test-cases/run-functional-tests.sh` to specify `test/test-cases/functional/ptf/` instead of `test/test-cases/functional/`. This may be superseded by related changes in https://github.com/Azure/DASH/pull/305 so we can adjust this PR to follow. Interim change here is required to ensure PTF doesn't "see" `test/test-cases/functional/saic`.

### CODE REVIEW

File: test/test-cases/functional/saic/sai_dpu_client_server_ptf.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{"DPU": [
```
[the remark this answers is not in the document]
Comment (chrispsommers) [reply test/test-cases/functional/saic/sai_dpu_client_server_ptf.json:1]: I'd like to defer that until later. I may propose a slight restructuring of scale and functional categories overall.

[the remark this answers is not in the document]
Comment (aputriax) [reply test/test-cases/functional/saic/sai_dpu_client_server_ptf.json:1]: ok. we just shouldn't forget about this.

File: test/test-cases/scale/saic/run-tests.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+PYTHONPATH=. pytest -sv $@
```
Comment (aputriax) [inline test/test-cases/scale/saic/run-tests.sh:3]: I would like to have a single entry point for all tests one day. Anyway all of them are going to use the same setup and the same connection protocols. Only the test content is different.

**Review by aputriax — APPROVED** [review]

File: test/test-cases/scale/saic/run-tests.sh (hunk 2 above)
Comment (chrispsommers) [reply test/test-cases/scale/saic/run-tests.sh:3]: @aputriax Agreed, I added an item to the DASH Test WG project tracker, which I invited you to. It's not quite the same as issues but good for tracking backlog items.

**Pull request merged**